ddmrp/models/stock_buffer.py

```diff
@@ -730,6 +730,59 @@ def _compute_main_supplier(self):
             else:
                 rec.main_supplier_id = False
 
+    def _compute_outside_dlt(self):
+        for rec in self:
+            pos = self.env["purchase.order"]
+            mos = self.env["mrp.production"]
+            picks = self.env["stock.picking"]
+            cut_date = rec._get_incoming_supply_date_limit()
+            locations = self.env["stock.location"].search(
+                [("id", "child_of", [rec.location_id.id])]
+            )
+            if rec.item_type == "purchased":
```

Comment:
Why isn't it always based on the stock move?
_search_stock_moves_incoming_domain is looking at moves only so why it differs here?
The PO date could differ from the move date if the supplier informs you afterwards of new delivery date

Reply:
So if I understand it well, instead of having a dynamic computation of the warning triangle, we base this one on the outside horizon incoming qty ?
I like this idea ? What do you need ?

Reply:
I agree and I like your suggestions. I'll adapt it

Reply:
There is one issue though and it is RFQ's as they don't yet have stock moves.